src/rust/lib/graph-editor/src/component/visualization.rs

```diff
@@ -0,0 +1,332 @@
+        let visible  = Cell::new(true);
+        let node     = display::object::Instance::new(&logger);
+
+        let data     = ContainerData {logger,events, visualization: content,size,position,visible,node};
+        let data     = Rc::new(data);
+        Self {data} . init_frp()
+    }
+
+    /// Update the content properties with the values from the `ContainerData`.
+    ///
+    /// Needs to called when a visualisation has been set.
+    fn update_visualisation_properties(&self) {
+        let size       = self.data.size.get();
+        let position   = self.data.position.get();
+
+        if let Some(vis) = self.data.visualization.borrow().as_ref() {
+            vis.content.set_size(size);
+            vis.content.set_position(position);
+        };
+    }
+
+    /// Set the visualization content.
+    pub fn set_visualisation(&self, visualization: Visualization) {
+        self.display_object().add_child(visualization.display_object());
+        self.data.visualization.replace(Some(visualization));
+        self.update_visualisation_properties();
+    }
+
+    fn init_frp(self) -> Self {
+        let network = &self.data.events.network;
+
+        frp::extend! { network
+            let weak_vis = self.downgrade();
+            def _f_show = self.data.events.show.map(move |_| {
+               if let Some(vis) = weak_vis.upgrade() {
+                    vis.set_visibility(true)
+               }
+            });
+
+            let weak_vis = self.downgrade();
+            def _f_hide = self.data.events.hide.map(move |_| {
+                if let Some(vis) = weak_vis.upgrade() {
+                    vis.set_visibility(false)
+               }
+            });
+
+            let weak_vis = self.downgrade();
+            def _f_toggle = self.data.events.hide.map(move |_| {
+                if let Some(vis) = weak_vis.upgrade() {
+                    vis.toggle_visibility()
+               }
+            });
+
+            let weak_vis = self.downgrade();
+            def _f_hide = self.data.events.update_visualization.map(move |content| {
+                if let Some(vis) = weak_vis.upgrade() {
+                    if let Some(content) = content.clone() {
+                        vis.set_visualisation(content);
+                    }
+                }
+            });
+
+            let weak_vis = self.downgrade();
+            def _f_hide = self.data.events.update_data.map(move |data| {
+                if let Some(vis) = weak_vis.upgrade() {
+                    vis.set_data(data.clone());
+                }
+            });
+        }
+        self
+    }
+
+    /// Toggle visibility on or off.
+    pub fn set_visibility(&self, visible: bool) {
+        self.data.visible.set(visible)  ;
+        if let Some(vis) = self.data.visualization.borrow().deref() {
+            // TODO use display object functionality
+            if visible {
+                vis.show();
+            } else {
+                vis.hide();
+            }
```

Comment:
minor api request - `vis` should have `set_visibility` method which will accept bool. It is a popular need to do something like that and such method would make the API nicer.